### PR TITLE
Let the view know about the review status of the PR

### LINKED ISSUE #473: Approve button should be disabled when PR is alreay approved
Issue Type: <b>Bug</b>

Enter Review mode for a PR

Type your'e comment (or not) and press "approve".

The button is still enabled and there's no indication that you've already approve this.

The UI should show the user if a PR was already approved/requested changes (possibly like GH does with the green_checkmark / red cross)

And if it was already approve it doesn't make much sense to still allow the user to "approve again" 

Extension version: 0.1.4
VS Code version: Code 1.27.1 (5944e81f3c46a3938a82c701f96d7a59b074cfdc, 2018-09-06T09:14:32.385Z)
OS version: Darwin x64 17.7.0

### PR DESCRIPTION
**Pull request by RishabhKothaari**
Fixes #473
The PR model must find the current user's review status for the PR and send it to the view.
Let me know if it requires any further changes.

### CODE REVIEW

**Review by alexr00 — CHANGES_REQUESTED** [review]
This PR is looking good! If the merge conflicts are resolved I will take another look.

**Review by alexr00 — APPROVED** [review]

**Review by alexr00 — CHANGES_REQUESTED** [review]
Accidentally hit approve.

**Comment by RishabhKothaari** [comment]
@alexr00 Thanks for reviewing it. I will resolve the conflicts. I hope there might have not been many changes since I last worked on VSCode PR

File: webviews/common/context.tsx [hunk 1]
```diff
@@ -158,6 +158,7 @@ export class PRContext {
 		});
 		state.reviewers = reviewers;
 		state.events = [...state.events.filter(e => (isReviewEvent(e) ? e.state !== 'PENDING' : e)), review];
+		state.reviewState = review.state;
```
Comment (alexr00) [inline webviews/common/context.tsx:161]: Is a check needed here to make sure that the review is from the current user?

**Review by alexr00 — CHANGES_REQUESTED** [review]
Thanks for the updates! Just one more question.

File: webviews/common/context.tsx (hunk 1 above)
Comment (RishabhKothaari) [reply webviews/common/context.tsx:161]: @alexr00 I did not see a need to check if the review is from the current user as this refers to the review state of the PR irrespective of who did it. It is required to enable/disable the Approve button on the overview of the PR. 
When the PR panel is shown it always refers to the review state from the current user for the PR
When the reviewer makes changes to the PR ```review``` always refers to the current user's review on the PR.
Please let me know if I am missing anything here, I am happy to make changes/updates

**Review by alexr00 — APPROVED** [review]
Looks good, thank you!

**Pull request merged**